### PR TITLE
refactor(API communication::add_resource)

### LINKED ISSUE #22: Refactor: use Promises for API communication
Currently we are using JQueries `$.ajax` to send requests to `ajax_product_descriptor` and handle the callbacks in `product_descriptor.config.js`.

For issue #20 we already implemented  **Promise** based communication for one endpoint, now we want to replace the other existing endpoints to work the same way.

## Goal
- no  usage of `api.call()` anymore
- get rid off `product_descriptor.config.js`, all callbacks are now handled in app.js via Promises

## Endpoints ToDo
- [x] 'add_resource'
- [x] 'bulk_translation_complete'
- [x] 'delete_resource'
- [x] 'generate_description'
- [x] 'get_materials'
- [x] 'get_metatags'
- [x] 'get_baseproducts'
- [x] 'get_components'
- [x] 'get_products'
- [x] 'set_hidden_tag'
- [x] 'save_resource'
- [x] 'save_product'
- [x] 'save_products'
- [x] 'save_resources'
- [x] 'translation_update'
- [x] 'validate_translation_update'
- [x] 'update_translation_requested'
- [x] 'translation_complete'
- [x] 'get_translation_status' => remove it out

Create a **new branch** based on `'bulk_translation_complete'` and work the list of

### PR DESCRIPTION
**Pull request by lalitaCF**
close #22

### CODE REVIEW

File: assets/app.js [hunk 1]
```diff
@@ -850,13 +850,25 @@ new Vue({
...
                 }
 
-                if (hasApi) {
-                    // clear product input
-                    api.app = this
-                    api.data = resource
-                    api.action = 'add_resource'
-                    api.call()
-                }
+                _this = this
```
Comment (Solomonic) [inline assets/app.js]: do we need `_this` ?

File: assets/app.js [hunk 2]
```diff
@@ -850,13 +850,25 @@ new Vue({
...
+                fetch(
+                    api_endpoint, {
```
Comment (Solomonic) [inline assets/app.js]: please use proper format
```
fetch(
  api_end_point,
  {
    method: 'POST',
    body: JSON.stringify(....)
  },
)
```

**Comment by adhocore** [comment]
> refactor(API communication::add_resource):wip

WIP (in caps) should be prefixed and not suffixed :)

File: assets/app.js [hunk 3]
```diff
@@ -430,10 +478,28 @@ new Vue({
             }
 
             if(this.rawMetatags==null){
-                api.app = this
-                api.action = 'get_metatags'
-                api.call()
-                return []
+                if(this.rawMetatags==null){
```
Comment (Solomonic) [inline assets/app.js]: remove double `if(this.rawMetatags==null){`

File: assets/app.js [hunk 4]
```diff
@@ -746,10 +924,28 @@ new Vue({
                 return []
             }
             if(this.rawMetatags==null){
-                api.app = this
-                api.action = 'get_metatags'
-                api.call()
-                return []
+                if(this.rawMetatags==null){
```
Comment (Solomonic) [inline assets/app.js]: remove double `if(this.rawMetatags==null){`

File: assets/app.js [hunk 5]
```diff
@@ -850,27 +1061,78 @@ new Vue({
...
+                                product: this.selectedDirtyProducts[0],
+                                languages: this.activeLanguagesId
+                            }
+                        })
+                    })
+                    .then(function (response) {
+                        return response.json()
+                    })
+                    .then(function (response) {
+                        if (typeof _this.modelCodeSuccess == 'undefined') {
```
Comment (Solomonic) [inline assets/app.js]: what is `modelCodeSuccess `?
It should be define in `data`

**Review by Solomonic — CHANGES_REQUESTED** [review]

**Pull request merged**